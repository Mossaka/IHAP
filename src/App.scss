@import './global';

.app {
  background-color: $bg-color;
}
