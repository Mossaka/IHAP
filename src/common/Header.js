import React from 'react';
import { Link, withRouter } from 'react-router-dom';
import logo from '../assets/logo.png';
import lever from '../assets/lever.png';
import User from './User';
import './Header.css';
import firebase from 'firebase';
import { Button, Col, Input } from 'reactstrap';


class Header extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      random: false,
      rotation: 0,
      keyword: ' ',
      loggedIn: false,
      showSignup: false, 
      ticketIDs: [],
      ticketDisplayed: '',
      numTickets: 0
    }
    this.handleLeverClick = this.handleLeverClick.bind(this);
    this.setRandom = this.setRandom.bind(this);
    this.setSearch = this.setSearch.bind(this);
    this.handleKeyPress = this.handleKeyPress.bind(this);
    this.search = this.search.bind(this);
    this.loadRandomTicket = this.loadRandomTicket.bind(this);

    // Save load ticket keys from database. 
    var tickets = firebase.database().ref('tickets'); 
    tickets.once('value').then((snapshot) => {
      var numTickets = snapshot.numChildren();
      var ticketKeys = []; 
      snapshot.forEach(((snapshot) => {
        var key = snapshot.key;
        ticketKeys.push(key);
      }));
      this.setState (
        {
          ticketIDs: ticketKeys, 
          numTickets: numTickets
        }
      )
    });
  }

  search() {
    this.props.history.push('/search/' + this.state.keyword);
  }

  handleKeyPress(e) {
    this.setState (
      {
        keyword: e.target.value,
      }
    );
    if(e.key === 'Enter') {
      this.search();
    }
  }

  
  handleLeverClick() {
    if (this.state.random === true) {
      this.setSearch();
    } else {
      this.setRandom();
    }
  }

  setRandom() {
    this.setState(
      {
        random: true,
        rotation: 180,
      }
    );
  }

  setSearch() {
    this.setState(
      {
        random: false,
        rotation: 0,
      }
    );
  }

  loadRandomTicket() {
    var ticketKey = this.state.ticketIDs[Math.floor(Math.random() * this.state.numTickets)];
    if(ticketKey !== this.state.ticketDisplayed) {
      console.log(ticketKey);
      this.props.history.push('/ticket/' + ticketKey);
      this.setState (
        {
          ticketDisplayed: ticketKey
        }
      )
    }    
  }


  render() {
    const searchOrButton = this.state.random ? (
      <div className="searchOrRandom">
        <Button className="searchOrRandomItem" color="secondary" onClick={this.loadRandomTicket} block >GET RANDOM TICKET</Button>
      </div>
    ) : (
      <div className="searchOrRandom">
        <Input className="searchOrRandomItem" type="search" name="search" placeholder="Search" onKeyUp={this.handleKeyPress}/>
        <Button color="secondary" onClick={this.search}>Search</Button>
      </div>
    ); 

    return (
      <nav id='nav'>
        <Col className="logo" xs="3">
          <Link to="/"><img src={logo} className="image" alt="IHAP Logo" /></Link>
        </Col>

        <Col className="center" xs="6">
          {searchOrButton}
          <div id="lever">
            <div>
              <img src={lever} alt="Lever" className="image clickable" style={{ transform: `rotate(${this.state.rotation}deg)` }} onClick={this.handleLeverClick} />
            </div>
            <div>
              <div className="clickable" id="random" onClick={this.setRandom}>Random</div>
              <div className="clickable" id="search" onClick={this.setSearch}>Search</div>
            </div>
          </div>
        </Col>

        <Col xs="3">
          <User />
        </Col>
      </nav>
    );
  }
}
export default withRouter(Header);
