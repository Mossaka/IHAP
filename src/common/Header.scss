.navbar {
  display: flex;
  justify-content: space-between;
  background-color: rgb(51, 51, 51);
  padding: 7px 15px 7px 15px;
  border-bottom: 2px solid lightgray;

  .logo:hover {
    text-decoration: none;
  }

  img {
    height: 80px;
  }

  .center {
    display: flex;
    justify-content: center;

    .lever {
      margin-left: 10px;
      display: flex;
      cursor: pointer;
      border: inset;
      border-radius: 5px;
      background: #333;

      img.true {
        transform: rotate(180deg);
      }

      .label {
        display: flex;
        flex-direction: column;
        justify-content: space-between;
        
       

        div {
          color: #DDD;
          padding-left: 3px;
          padding-right: 3px;
          -webkit-user-select: none; /* Safari */        
          -moz-user-select: none; /* Firefox */
          -ms-user-select: none; /* IE10+/Edge */
          user-select: none; /* Standard */
        }
      }
    }

    .searchOrRandom {
      display: flex;
      flex-grow: 1;
      justify-content: center;
      align-items: center;

      input {
        border: 2px solid steelblue;
      }

      .searchButton {
        background-color: steelblue;
        border: 2px solid steelblue;
      }

      .btn:hover {
        background-color:#2A6A90;
        border: 2px solid #2A6A90;
      } 
    }
  }
}
