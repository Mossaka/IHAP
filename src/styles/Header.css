.image {
    height: 80px;
}
.upsidedownImage {
    height: 80px;
    transform: rotate(180deg);
}

.button {
    width: 100px; 
}

.left {
    float:left;
}
.right {
    float:right; 
}