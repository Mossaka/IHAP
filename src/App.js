import React from 'react';
import { Route, Switch } from 'react-router-dom';
import Header from './common/Header';
import HomePage from './home/HomePage';
import ProfilePage from './profile/ProfilePage';
import TicketPage from './ticket/TicketPage';
import SearchPage from './searchResults/SearchPage';

export default class App extends React.Component {
  render() {
    return (
      <React.Fragment>
        <Header/>
        <Switch>
          <Route path="/ticket/:id" component={TicketPage} />
          <Route path="/search/:keyword" component={SearchPage} />
          <Route path="/profile/:id" component={ProfilePage} />
          <Route component={HomePage} />
        </Switch>
      </React.Fragment>
    );
  }
}
