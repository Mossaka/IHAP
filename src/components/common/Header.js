import React, {PropTypes} from 'react';
import { Link, IndexLink } from 'react-router';
import logo from '../../assets/logo.png';
import lever from '../../assets/lever.png';
import '../../styles/Header.css';
import { Button } from 'reactstrap';


const Header = ({loading}) => {

  return (
    <nav>
      <span className="left logo">
      <IndexLink to="/" activeClassName="active"><img src={logo} className="image" alt="IHAP Logo" /></IndexLink>
      </span>

      <span className="center"> 
      <Link to="/tickets" activeClassName="active">Tickets</Link>
      {" | "}
      <Link to="/profile" activeClassName="active">Profile</Link>
        <span>
        <img src={lever} className="image" alt="Lever" />
        <img src={lever} className="upsidedownImage" alt="Lever" />
        </span>
        <span id="lever-options">
          Random
          Search
        </span>
      </span>
      
      <span className="right user">
        <Link to="/signin" activeClassName="active"><Button className="button" color="primary" size="sm">Log in</Button></Link>
        <br/ >
        <Link to="/signup" activeClassName="active"><Button className="button" color="primary" size="sm">Sign up</Button></Link>
      </span>
    </nav>
  );
};

export default Header;
