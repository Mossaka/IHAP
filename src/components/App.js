import React, { Component } from 'react';
import logo from '../assets/logo.png';
import '../styles/App.css';
import Header from './common/Header';

class App extends Component {
  render() {
    return (
      <div className="App">
        <header className="App-header">
          <Header />
          <h1 className="App-title">Welcome to IHAP</h1>
        </header>
        {this.props.children}
      </div>
    );
  }
}

export default App;
