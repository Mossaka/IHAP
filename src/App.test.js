import React from 'react';
import renderer from 'react-test-renderer';
import {shallow, mount, render, configure} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';
import ReactDOM from 'react-dom';
import App from './components/App';
import Header from './components/common/Header';
import Adapter from 'enzyme-adapter-react-16';
import { Link } from 'react-router-dom';

configure({ adapter: new Adapter() });

describe('Basic Functionality', () => {
  
  // it('renders without crashing', () => {
  //   const div = document.createElement('div');
  //   ReactDOM.render(<App />, div);
  //   ReactDOM.unmountComponentAtNode(div);
  // });

  it('cmon', () => {
    expect(1 + 2).toBe(3);
  })
})

describe('Header', () =>{
  // check header
  it('header defined', () => {
    expect(Header).toBeDefined();
  })

  // renders
  it('header renders', () => {
    expect(shallow(<Header/>)).toMatchSnapshot();
  });

  // sign in button
  it('header has sign in', () => {
    expect(shallow(<Header/>).find('Link[to="/signin"]').length).toEqual(1);
  });

  // random lever
  it('random lever toggles', () => {
    const wrapper = shallow(<Header/>);
    expect(wrapper.state('random')).toBe(false);
    wrapper.find('img[alt="Lever"]').last().simulate('click');
    expect(wrapper.state('random')).toBe(true);
    wrapper.find('img[alt="Lever"]').last().simulate('click');
    expect(wrapper.state('random')).toBe(false);
  });
})