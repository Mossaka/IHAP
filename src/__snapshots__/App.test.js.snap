// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header header renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Col
          className="logo"
          tag="div"
          widths={
                    Array [
                              "xs",
                              "sm",
                              "md",
                              "lg",
                              "xl",
                            ]
          }
          xs="3"
>
          <Link
                    replace={false}
                    to="/"
          >
                    <img
                              alt="IHAP Logo"
                              className="image"
                              src="logo.png"
                    />
          </Link>
</Col>,
        <Col
          className="center"
          tag="div"
          widths={
                    Array [
                              "xs",
                              "sm",
                              "md",
                              "lg",
                              "xl",
                            ]
          }
          xs="auto"
>
          <div>
                    <Input
                              name="search"
                              placeholder="Search"
                              type="search"
                    />
                    <Button
                              color="primary"
                              tag="button"
                    >
                              SEARCH
                    </Button>
          </div>
          <div
                    id="lever"
          >
                    <div>
                              <img
                                        alt="Lever"
                                        className="image clickable"
                                        onClick={[Function]}
                                        src="lever.png"
                                        style={
                                                  Object {
                                                            "transform": "rotate(0deg)",
                                                          }
                                        }
                              />
                    </div>
                    <div>
                              <div
                                        className="clickable"
                                        id="random"
                                        onClick={[Function]}
                              >
                                        Random
                              </div>
                              <div
                                        className="clickable"
                                        id="search"
                                        onClick={[Function]}
                              >
                                        Search
                              </div>
                    </div>
          </div>
</Col>,
        <Col
          className="user"
          tag="div"
          widths={
                    Array [
                              "xs",
                              "sm",
                              "md",
                              "lg",
                              "xl",
                            ]
          }
          xs="3"
>
          <Link
                    activeClassName="active"
                    replace={false}
                    to="/signin"
          >
                    <Button
                              className="button"
                              color="primary"
                              size="sm"
                              tag="button"
                    >
                              Log in
                    </Button>
          </Link>
          <br />
          <Link
                    activeClassName="active"
                    replace={false}
                    to="/signup"
          >
                    <Button
                              className="button"
                              color="primary"
                              size="sm"
                              tag="button"
                    >
                              Sign up
                    </Button>
          </Link>
</Col>,
      ],
      "id": "nav",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Link
            replace={false}
            to="/"
>
            <img
                        alt="IHAP Logo"
                        className="image"
                        src="logo.png"
            />
</Link>,
          "className": "logo",
          "tag": "div",
          "widths": Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ],
          "xs": "3",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <img
              alt="IHAP Logo"
              className="image"
              src="logo.png"
/>,
            "replace": false,
            "to": "/",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "IHAP Logo",
              "className": "image",
              "src": "logo.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <div>
              <Input
                            name="search"
                            placeholder="Search"
                            type="search"
              />
              <Button
                            color="primary"
                            tag="button"
              >
                            SEARCH
              </Button>
</div>,
            <div
              id="lever"
>
              <div>
                            <img
                                          alt="Lever"
                                          className="image clickable"
                                          onClick={[Function]}
                                          src="lever.png"
                                          style={
                                                        Object {
                                                                      "transform": "rotate(0deg)",
                                                                    }
                                          }
                            />
              </div>
              <div>
                            <div
                                          className="clickable"
                                          id="random"
                                          onClick={[Function]}
                            >
                                          Random
                            </div>
                            <div
                                          className="clickable"
                                          id="search"
                                          onClick={[Function]}
                            >
                                          Search
                            </div>
              </div>
</div>,
          ],
          "className": "center",
          "tag": "div",
          "widths": Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ],
          "xs": "auto",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Input
                  name="search"
                  placeholder="Search"
                  type="search"
/>,
                <Button
                  color="primary"
                  tag="button"
>
                  SEARCH
</Button>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "name": "search",
                  "placeholder": "Search",
                  "type": "search",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "SEARCH",
                  "color": "primary",
                  "tag": "button",
                },
                "ref": null,
                "rendered": "SEARCH",
                "type": [Function],
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div>
                  <img
                                    alt="Lever"
                                    className="image clickable"
                                    onClick={[Function]}
                                    src="lever.png"
                                    style={
                                                      Object {
                                                                        "transform": "rotate(0deg)",
                                                                      }
                                    }
                  />
</div>,
                <div>
                  <div
                                    className="clickable"
                                    id="random"
                                    onClick={[Function]}
                  >
                                    Random
                  </div>
                  <div
                                    className="clickable"
                                    id="search"
                                    onClick={[Function]}
                  >
                                    Search
                  </div>
</div>,
              ],
              "id": "lever",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <img
                    alt="Lever"
                    className="image clickable"
                    onClick={[Function]}
                    src="lever.png"
                    style={
                                        Object {
                                                            "transform": "rotate(0deg)",
                                                          }
                    }
/>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "Lever",
                    "className": "image clickable",
                    "onClick": [Function],
                    "src": "lever.png",
                    "style": Object {
                      "transform": "rotate(0deg)",
                    },
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <div
                      className="clickable"
                      id="random"
                      onClick={[Function]}
>
                      Random
</div>,
                    <div
                      className="clickable"
                      id="search"
                      onClick={[Function]}
>
                      Search
</div>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Random",
                      "className": "clickable",
                      "id": "random",
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": "Random",
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Search",
                      "className": "clickable",
                      "id": "search",
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": "Search",
                    "type": "div",
                  },
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Link
              activeClassName="active"
              replace={false}
              to="/signin"
>
              <Button
                            className="button"
                            color="primary"
                            size="sm"
                            tag="button"
              >
                            Log in
              </Button>
</Link>,
            <br />,
            <Link
              activeClassName="active"
              replace={false}
              to="/signup"
>
              <Button
                            className="button"
                            color="primary"
                            size="sm"
                            tag="button"
              >
                            Sign up
              </Button>
</Link>,
          ],
          "className": "user",
          "tag": "div",
          "widths": Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ],
          "xs": "3",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "activeClassName": "active",
              "children": <Button
                className="button"
                color="primary"
                size="sm"
                tag="button"
>
                Log in
</Button>,
              "replace": false,
              "to": "/signin",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Log in",
                "className": "button",
                "color": "primary",
                "size": "sm",
                "tag": "button",
              },
              "ref": null,
              "rendered": "Log in",
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "activeClassName": "active",
              "children": <Button
                className="button"
                color="primary"
                size="sm"
                tag="button"
>
                Sign up
</Button>,
              "replace": false,
              "to": "/signup",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Sign up",
                "className": "button",
                "color": "primary",
                "size": "sm",
                "tag": "button",
              },
              "ref": null,
              "rendered": "Sign up",
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "nav",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Col
            className="logo"
            tag="div"
            widths={
                        Array [
                                    "xs",
                                    "sm",
                                    "md",
                                    "lg",
                                    "xl",
                                  ]
            }
            xs="3"
>
            <Link
                        replace={false}
                        to="/"
            >
                        <img
                                    alt="IHAP Logo"
                                    className="image"
                                    src="logo.png"
                        />
            </Link>
</Col>,
          <Col
            className="center"
            tag="div"
            widths={
                        Array [
                                    "xs",
                                    "sm",
                                    "md",
                                    "lg",
                                    "xl",
                                  ]
            }
            xs="auto"
>
            <div>
                        <Input
                                    name="search"
                                    placeholder="Search"
                                    type="search"
                        />
                        <Button
                                    color="primary"
                                    tag="button"
                        >
                                    SEARCH
                        </Button>
            </div>
            <div
                        id="lever"
            >
                        <div>
                                    <img
                                                alt="Lever"
                                                className="image clickable"
                                                onClick={[Function]}
                                                src="lever.png"
                                                style={
                                                            Object {
                                                                        "transform": "rotate(0deg)",
                                                                      }
                                                }
                                    />
                        </div>
                        <div>
                                    <div
                                                className="clickable"
                                                id="random"
                                                onClick={[Function]}
                                    >
                                                Random
                                    </div>
                                    <div
                                                className="clickable"
                                                id="search"
                                                onClick={[Function]}
                                    >
                                                Search
                                    </div>
                        </div>
            </div>
</Col>,
          <Col
            className="user"
            tag="div"
            widths={
                        Array [
                                    "xs",
                                    "sm",
                                    "md",
                                    "lg",
                                    "xl",
                                  ]
            }
            xs="3"
>
            <Link
                        activeClassName="active"
                        replace={false}
                        to="/signin"
            >
                        <Button
                                    className="button"
                                    color="primary"
                                    size="sm"
                                    tag="button"
                        >
                                    Log in
                        </Button>
            </Link>
            <br />
            <Link
                        activeClassName="active"
                        replace={false}
                        to="/signup"
            >
                        <Button
                                    className="button"
                                    color="primary"
                                    size="sm"
                                    tag="button"
                        >
                                    Sign up
                        </Button>
            </Link>
</Col>,
        ],
        "id": "nav",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Link
              replace={false}
              to="/"
>
              <img
                            alt="IHAP Logo"
                            className="image"
                            src="logo.png"
              />
</Link>,
            "className": "logo",
            "tag": "div",
            "widths": Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ],
            "xs": "3",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <img
                alt="IHAP Logo"
                className="image"
                src="logo.png"
/>,
              "replace": false,
              "to": "/",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "IHAP Logo",
                "className": "image",
                "src": "logo.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <div>
                <Input
                                name="search"
                                placeholder="Search"
                                type="search"
                />
                <Button
                                color="primary"
                                tag="button"
                >
                                SEARCH
                </Button>
</div>,
              <div
                id="lever"
>
                <div>
                                <img
                                                alt="Lever"
                                                className="image clickable"
                                                onClick={[Function]}
                                                src="lever.png"
                                                style={
                                                                Object {
                                                                                "transform": "rotate(0deg)",
                                                                              }
                                                }
                                />
                </div>
                <div>
                                <div
                                                className="clickable"
                                                id="random"
                                                onClick={[Function]}
                                >
                                                Random
                                </div>
                                <div
                                                className="clickable"
                                                id="search"
                                                onClick={[Function]}
                                >
                                                Search
                                </div>
                </div>
</div>,
            ],
            "className": "center",
            "tag": "div",
            "widths": Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ],
            "xs": "auto",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Input
                    name="search"
                    placeholder="Search"
                    type="search"
/>,
                  <Button
                    color="primary"
                    tag="button"
>
                    SEARCH
</Button>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "name": "search",
                    "placeholder": "Search",
                    "type": "search",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "SEARCH",
                    "color": "primary",
                    "tag": "button",
                  },
                  "ref": null,
                  "rendered": "SEARCH",
                  "type": [Function],
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div>
                    <img
                                        alt="Lever"
                                        className="image clickable"
                                        onClick={[Function]}
                                        src="lever.png"
                                        style={
                                                            Object {
                                                                                "transform": "rotate(0deg)",
                                                                              }
                                        }
                    />
</div>,
                  <div>
                    <div
                                        className="clickable"
                                        id="random"
                                        onClick={[Function]}
                    >
                                        Random
                    </div>
                    <div
                                        className="clickable"
                                        id="search"
                                        onClick={[Function]}
                    >
                                        Search
                    </div>
</div>,
                ],
                "id": "lever",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <img
                      alt="Lever"
                      className="image clickable"
                      onClick={[Function]}
                      src="lever.png"
                      style={
                                            Object {
                                                                  "transform": "rotate(0deg)",
                                                                }
                      }
/>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "alt": "Lever",
                      "className": "image clickable",
                      "onClick": [Function],
                      "src": "lever.png",
                      "style": Object {
                        "transform": "rotate(0deg)",
                      },
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "img",
                  },
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <div
                        className="clickable"
                        id="random"
                        onClick={[Function]}
>
                        Random
</div>,
                      <div
                        className="clickable"
                        id="search"
                        onClick={[Function]}
>
                        Search
</div>,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Random",
                        "className": "clickable",
                        "id": "random",
                        "onClick": [Function],
                      },
                      "ref": null,
                      "rendered": "Random",
                      "type": "div",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Search",
                        "className": "clickable",
                        "id": "search",
                        "onClick": [Function],
                      },
                      "ref": null,
                      "rendered": "Search",
                      "type": "div",
                    },
                  ],
                  "type": "div",
                },
              ],
              "type": "div",
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <Link
                activeClassName="active"
                replace={false}
                to="/signin"
>
                <Button
                                className="button"
                                color="primary"
                                size="sm"
                                tag="button"
                >
                                Log in
                </Button>
</Link>,
              <br />,
              <Link
                activeClassName="active"
                replace={false}
                to="/signup"
>
                <Button
                                className="button"
                                color="primary"
                                size="sm"
                                tag="button"
                >
                                Sign up
                </Button>
</Link>,
            ],
            "className": "user",
            "tag": "div",
            "widths": Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ],
            "xs": "3",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "activeClassName": "active",
                "children": <Button
                  className="button"
                  color="primary"
                  size="sm"
                  tag="button"
>
                  Log in
</Button>,
                "replace": false,
                "to": "/signin",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Log in",
                  "className": "button",
                  "color": "primary",
                  "size": "sm",
                  "tag": "button",
                },
                "ref": null,
                "rendered": "Log in",
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "activeClassName": "active",
                "children": <Button
                  className="button"
                  color="primary"
                  size="sm"
                  tag="button"
>
                  Sign up
</Button>,
                "replace": false,
                "to": "/signup",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Sign up",
                  "className": "button",
                  "color": "primary",
                  "size": "sm",
                  "tag": "button",
                },
                "ref": null,
                "rendered": "Sign up",
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "nav",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
